packaging/python/fix_demo_libnrnmech.sh

```diff
@@ -9,7 +9,11 @@ set -ex
 
 instdir=$1
 REL_RPATH=$2
-libnrnmech_dir=$instdir/share/nrn/demo/release/x86_64
+if [[ `uname -m` == 'arm64' ]]; then
+  libnrnmech_dir=$instdir/share/nrn/demo/release/arm64
+else
+  libnrnmech_dir=$instdir/share/nrn/demo/release/x86_64
+fi
```

Comment:
``` 
libnrnmech_dir=$instdir/share/nrn/demo/release/`uname -m`
```
?

Reply:
In the context of Arm64 I am thinking about @tristan0x's suggestion because on circleci-arm we now have `aarch64` 😒. Looking at [CMake docs](https://cmake.org/cmake/help/latest/variable/CMAKE_HOST_SYSTEM_PROCESSOR.html):

![image](https://user-images.githubusercontent.com/666852/156664109-938282fa-bd90-437f-b337-8fdb22d56d5e.png)

So this definitely needs more care!

Reply:
Tbh, I'd like to see the platform disappear entirely.

Reply:
I completely agree, but that would be for next (major) release. But as we started doing fixes for Apple M1 and AArch64, we need to such changes anyway. (I will PR circle-ci soon)